src/fides/api/alembic/migrations/versions/14acee6f5459_translation_data.py

```diff
@@ -0,0 +1,827 @@
+                )
+                continue
+
+            # revert to the "raw" config if we haven't flagged a config as needing migration
+            # in practice, we should always have a reconciled record here that's flagged for migration,
+            # but this covers our bases in case anyone has e.g. deleted a previous OOB notice.
+            if reconciled_experience_config.get("needs_migration", False):
+                logger.info(
+                    f"Migrating reconciled experience config {reconciled_experience_config['id']}"
+                )
+                experience_config_type = reconciled_experience_config
+            else:
+                logger.info(
+                    f"Creating raw experience config {raw_experience_config['id']}"
+                )
+                experience_config_type = raw_experience_config
+
+            create_new_experience_config(experience_config_type)
+            create_applicable_experiences(experience_config_type)
+            link_notices_to_experiences(experience_config_type)
+
+        # experience translations are then created based on the new experience config records
+        create_experience_translations(bind)
+
+
+def migrate_notices(bind):
+    """
+    Migrates existing PrivacyNotice records to the new, translation-based data model.
+
+    Notice keys are not impacted. Most existing notice data is moved onto a linked NoticeTranslation record,
+    which is assumed to be english-language as a default. If the existing notice text/content is _not_ english-language,
+    then users will need to *manually* adjust the language on the created translation record.
+
+    The existing PrivacyNoticeHistory records are adjusted to point to the associated NoticeTranslation record,
+    as is expected in the new data model.
+    """
+
+    get_notices_query = text(
+        """SELECT id from privacynotice;
+         """
+    )
+
+    create_notice_translation_query = text(
+        """
+        INSERT INTO noticetranslation (id, privacy_notice_id, language, title, description, created_at, updated_at)
+        SELECT :record_id, :original_notice_id, :language, name, description, created_at, updated_at
+        FROM privacynotice
+        WHERE id = :original_notice_id
+        """
+    )
+    for res in bind.execute(get_notices_query):
+        # Create a Notice Translation for each notice, setting to english as a default
+        bind.execute(
+            create_notice_translation_query,
+            {
+                "record_id": generate_record_id("pri"),
+                "language": "en",
+                "original_notice_id": res["id"],
+            },
+        )
+
+    # Add a FK from the Notice History back to the Notice Translation
+    link_notice_history_to_translation_id = text(
+        """
+        UPDATE privacynoticehistory
+        SET translation_id = noticetranslation.id, title = privacynoticehistory.name, language = 'en'
```

Comment:
So for french customers, they will have to go in and update the language where we've said this is "en".  However, "en" is the identifier for all intents and purposes.  I think they would have to add their french translations, and delete the other translations that are also in french but labeled "english", unless we update this at the db level.

Reply:
Note this is potentially inaccurate for some customers on historical records.  We don't have to backfill it anymore, but we'd likely want it backfilled for the notice history in use at least, so this is fine.